common/src/worldgen.rs

```diff
@@ -167,154 +169,132 @@ impl ChunkParams {
+                    };
+
+                    let voxel_mat = if dist >= 0.0 {
+                        VoronoiInfo::terraingen_voronoi(elev, rain, temp, dist)
+                    } else {
+                        Material::Void
+                    };
+
+                    voxels.data_mut(self.dimension)[index(self.dimension, coords)] = voxel_mat;
                 }
-            } else if (slope_mod >= 3_f64) && (slope_mod <= 4_f64) {
-                voxel_mat = match voxel_mat {
-                    Material::Flowergrass => Material::Bigflowergrass,
-                    Material::Greystone => Material::Blackstone,
-                    _ => voxel_mat,
-                };
-                max_e = elev - 0.25;
-            } else {
-                max_e = elev;
             }
-        } else {
-            max_e = elev;
-        }
-
-        let voxel_elevation = self.surface.distance_to_chunk(self.chunk, &center);
-        if voxel_elevation >= max_e / ELEVATION_SCALE {
-            voxel_mat = Material::Void;
         }
-
-        voxel_mat
     }
 
-    fn generate_road(&self, center: na::Vector3<f64>) -> Option<Material> {
+    /// Places a road along the guiding plane.
+    fn generate_road(&self, voxels: &mut VoxelData) {
         let plane = -Plane::from(Side::B);
 
-        let horizontal_distance = plane.distance_to_chunk(self.chunk, &center);
-        let elevation = self.surface.distance_to_chunk(self.chunk, &center);
-        if horizontal_distance > 0.3 {
-            return None;
-        }
-        if elevation < 0.075 {
-            // Surface
-            Some(if horizontal_distance < 0.15 {
-                // Inner
-                Material::WhiteBrick
-            } else {
-                // Outer
-                Material::GreyBrick
-            })
-        } else if elevation < 0.9 {
-            Some(Material::Void) // Tunnel
-        } else {
-            None
+        for z in 0..self.dimension {
+            for y in 0..self.dimension {
+                for x in 0..self.dimension {
+                    let coords = na::Vector3::new(x, y, z);
+                    let center = voxel_center(self.dimension, coords);
+                    let horizontal_distance = plane.distance_to_chunk(self.chunk, &center);
+                    let elevation = self.surface.distance_to_chunk(self.chunk, &center);
+
+                    let mut mat: Material = Material::Void;
+                    if horizontal_distance > 0.3 {
+                        //mat = Material::Void;
+                        continue;
+                    }
+                    if elevation < 0.075 {
+                        // Surface
+                        if horizontal_distance < 0.15 {
+                            // Inner
+                            mat = Material::WhiteBrick;
+                        } else {
+                            // Outer
+                            mat = Material::GreyBrick;
+                        }
```

Comment:
This whole clause looks like a no-op; `mat` is not read from.

Reply:
It can't be a no-op---running the application for the PR branch shows that the road still exists, and the generate_road function is the only place that WhiteBrick and GreyBrick materials are used. The logic was expressed in a confusing way, so I've tidied it up and made it short circuit in that function and in generate_road_support.